Create minimap from player position

iFuSiiOnzZ opened the pull request:
The idea behind this is to generate a minimap of the track using the player driving line.

iFuSiiOnzZ committed:
Crate minimap from driving line

iFuSiiOnzZ committed:
Add option to rotate the map

iFuSiiOnzZ committed:
Add map information data structure

This information will be use to map cars positions with minimap

iFuSiiOnzZ committed:
Refactor the work that does the overlay and the tracking job

iFuSiiOnzZ committed:
Draw player and other on the minimap

iFuSiiOnzZ committed:
Tweak refresh rate and change  distance threshold

iFuSiiOnzZ committed:
Simplify drawing logic and add black border for car dots

iFuSiiOnzZ committed:
Use only lap number to start and finish the track mapping

iFuSiiOnzZ committed:
Scale based on the maximum width

iFuSiiOnzZ committed:
Better mapping between the map and the cars and add car identifier

iFuSiiOnzZ committed:
Fix track mapping, get player position instead of index 0

iFuSiiOnzZ committed:
Replace magic numbers with variables

RiddleTime commented on Race_Element.HUD.ACC/Overlays/Driving/TrackMap/TrackMapOverlay.cs:
```diff
@@ -13,57 +25,232 @@ namespace RaceElement.HUD.ACC.Overlays.Driving.OverlayTrackMap;
+        var w = (float)Math.Sqrt((boundaries.Right - boundaries.Left) * (boundaries.Right - boundaries.Left));
+        var  h = (float)Math.Sqrt((boundaries.Top - boundaries.Bottom) * (boundaries.Top - boundaries.Bottom));
+
+        var carsAndTrack = new Bitmap((int)(w + _margin), (int)(h + _margin));
```
`new Bitmap(width, height, PixelFormat.Format32bppPArgb)`
This format is most effecient.

iFuSiiOnzZ committed:
Show car number instead of identifier

iFuSiiOnzZ committed:
On a new session trigger map tracking job

iFuSiiOnzZ committed:
Fix crash online caused by accessing wrong index

iFuSiiOnzZ committed:
Show car number only if car is on the track

iFuSiiOnzZ committed:
Clear track points on new session notification

iFuSiiOnzZ committed:
Add blue and orange color for lapped cars

iFuSiiOnzZ committed:
Take into account the distance traveled to lap or when you are being lapped

RiddleTime committed:
Update EntryListTracker.cs

use Dictionary.TryGetValue to get car data in the entry list tracker

RiddleTime committed:
Update TrackMapCreationJob.cs

ensure both flestream and binrary writer will be disposed at the end of the method body

RiddleTime committed:
Update TrackMapCreationJob.cs

Same like for commit 93b3257dbf44b529d5e37a11dff971247b223f2d
But for reading the .bin files

iFuSiiOnzZ committed:
Change variable declaration from 'var' to 'using <vat_type>' (automativally disposes when it goes out of scope)

iFuSiiOnzZ committed:
Merge branch 'RiddleTime:dev' into dev

iFuSiiOnzZ committed:
Refactor the drawing method by adding its own class

- Car number have transparent background (easier to read)
- Player car is always draw on the of the others
